### ISSUE TITLE
Implement floccus app in NextCloud so batch operations can be done locally

### ISSUE THREAD

**Issue by skewty** [marcelklehr/floccus#189]
I am guessing the slow (synchronous?  multiple TCP round trips) is due to the Bookmarks app in Nextcloud.

A very lightweight Floccus app installed into Nextcloud could allow us to batch many operations into a single blob and send them off compressed.

Granted this is quite out of scope, but waiting 30 minutes for my bookmarks to sync (local 1 Gbit connection to otherwise Idle Nextcloud server) (I don't know if it is jammed up or still working away) is silly.  

Your thoughts?

**Comment by marcelklehr** [comment]
I originally wanted to run the API requests in parallel, but it turned out not to be a good idea. It might be possible to sort out the differences in one go and then run all the server mutations in parallel.

In any case, you could try the WebDAV adapter, which updates the server in one go and should be much faster in your case.

**Comment by marcelklehr** [comment]
I've implemented this and will make available a beta version for testing, shortly. In my tests this cut down initial sync time of 4000 bookmarks from 30+ minues to 7min.
